client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -557,6 +562,11 @@ public void flush() throws SegmentSealedException {
                 }
             } catch (Exception e) {
                 failConnection(e);
+                if (e instanceof RetriesExhaustedException) {
+                    //throw an exception to the external world that the flush failed due to RetriesExhaustedException
+                    Exceptions.sneakyThrow(e);
```

Comment:
The typical pattern here is to do:
` throw Exceptions.sneakyThrow(e);`
For this reason the signature of `sneakyThrow` returns an exception even though the function never returns. Calling it in this way makes it clear visually that an exception will be thrown from this line and makes static analysis tools like IDEs better able to parse what is going on.

Reply:
fixed it.